renku/service/views/datasets.py

```diff
@@ -0,0 +1,325 @@
+              schema:
+                type: string
+              required: true
+
+        responses:
+            result:
+                description: |
+                  Contains `datasets` key with
+                  list of all datasets from a specified project.
+
+            error:
+                description: |
+                  Contains `code` and `message`
+                  describing the error.
+
+    """
+    if request.method == 'GET':
+        project_id = request.args.get('project_id')
+        pwd = validate_project_id(project_id)
+        if not pwd:
+            return jsonify(status=2, message='invalid project_id argument')
+
+        cwd = os.getcwd()
+        os.chdir(str(pwd))
+
+        datasets = [{
+            'id': ds['_label'],
+            'name': ds['name'],
+            'version': ds['version'],
+            'tags': ds['tags']
+        } for ds in json.loads(dataset_parent(None, 'data', 'json-ld'))]
+
+        os.chdir(cwd)
+        return jsonify(datasets=datasets)
+
+
+@requires_identity
+def list_dataset_files_view(user):
+    """
+    List all dataset files in specified project id.
+
+    ---
+    get:
+        tags:
+        - "datasets"
+
+        summary: List all dataset files within a project.
+        description: List all dataset files within a project.
+
+        parameters:
+            - name: project_id
+              in: query
+              description: Project identifier.
+              schema:
+                type: string
+              required: true
+
+            - name: dataset_id
+              in: query
+              description: Dataset identifier.
+              schema:
+                type: string
+              required: true
+
+        responses:
+            result:
+                description: |
+                  Contains `datasets` key with
+                  list of all datasets from a specified project.
+
+            error:
+                description: |
+                  Contains `code` and `message`
+                  describing the error.
+
+    """
+    if request.method == 'GET':
+        dataset_name = request.args.get('dataset_id')
+        project_id = request.args.get('project_id')
+        pwd = validate_project_id(project_id)
+        if not pwd:
+            return jsonify(status=2, message='invalid project_id argument')
+
+        cwd = os.getcwd()
+        os.chdir(str(pwd))
+
+        files = [
+            ds for ds in
+            json.loads(list_files(dataset_name, None, None, None, 'json-ld'))
+        ]
+
+        os.chdir(cwd)
+        return jsonify(dataset=dataset_name, files=files)
+
+
+@requires_identity
+def add_file_to_dataset_view(user):
+    """
+    Add uploaded file to cloned repository.
+
+    ---
+    get:
+        tags:
+        - "datasets"
+
+        summary: Add a file from cache to cloned repository.
+        description: Add a file from cache to cloned repository
+
+        parameters:
+            - name: dataset_name
```

Comment:
Because this is not what the interface supports. When you add a file, you specify a dataset name and not dataset id (which we can argue that is broken due to DOI's in there).